### DISCUSSION TITLE
install error

### DISCUSSION THREAD

**Discussion by jwx123** [graphistry/graph-app-kit#87] [Q&A]
I following the readme to install graph-app-kit ,occurence the error
Step 15/27 : RUN --mount=type=cache,target=/root/.cache     { source activate rapids || echo ok ; } && pip install -r requirements-system.txt
ERROR: Service 'streamlit' failed to build: the --mount option requires BuildKit. Refer to https://docs.docker.com/go/buildkit/ to learn how to build images with BuildKit enabled
I have run  "export DOCKER_BUILDKIT=1 export COMPOSE_DOCKER_CLI_BUILD=1" before docker-compose build

**Comment by lmeyerov** [comment]
That's strange...

* can you confirm `env` reports those name/values ?
* what os --linux, osx, windows?

Some ideas:

* `docker compose build` should auto-enable buildkit nowadays
* in your global docker config (`daemon.json`), set `{ "features": { "buildkit": true }, ...}`, and restart docker

**Comment by jwx123** [comment]
I solved the question,but the installation process is as follows，can't open localhost:8051 after run docker-compose up -d 
![image](https://user-images.githubusercontent.com/44939202/196843034-bda19e78-355a-440e-b94f-9b41db20c3ac.png)

**Reply by kkarthikvk** [reply]
Hi,
I'm also reproducing the same issue. 
I also tried what @lmeyerov suggested. Created a daemon.json file with `{
  "features": {
    "buildkit" : true
  }
}`
Restarted docker. Yet the error code persists.
Could you eleborate on how you resolved the issue?

**Reply by lmeyerov** [reply]
@kkarthikvk @jwx123  any chance there is a `sudo ...` call happening that isn't getting the env?

**Reply by kkarthikvk** [reply]
@lmeyerov yeah I guess. I need sudo privilege to run docker. 
While running `sudo docker-compose build` getting the error.

**Detailed information regarding the issue :**

**Versions:**
Ubuntu focal 20.04
Docker version `Docker version 22.06.0-beta.0, build 3e9117b`
Docker-compose version `docker-compose version 1.21.0, build 5920eb0`

**To reproduce Error:**
After cloning graph-app-kit repo, 
cd to ../src/docker/ 
enable docker buildkit by "export DOCKER_BUILDKIT=1 && export COMPOSE_DOCKER_CLI_BUILD=1"

**Getting the below error while running:**
`sudo docker-compose build`
**Error:**
`Step 15/27 : RUN --mount=type=cache,target=/root/.cache { source activate rapids || echo ok ; } && pip install -r requirements-system.txt 
ERROR: Service 'streamlit' failed to build: the --mount option requires BuildKit. Refer to https://docs.docker.com/go/buildkit/ to learn how to build images with BuildKit enabled`

**Tried solutions from web:**
Tried  Creating a daemon.json file in /etc/docker/daemon.json with `{ "features": { "buildkit" : true } }`
Restarted docker also reloaded daemon using systemctl commands. Yet the error code persists.

**Reply by lmeyerov** [reply]
@kkarthikvk  yeah maybe needs to be `sudo DOCKER_BUILDKIT=1 ...` so all inside the `sudo`

Also I noticed docker-compose 1.21 is from 2018, so maybe that's pre-buildkit?

**Reply by kkarthikvk** [reply]
@lmeyerov all inside `sudo`  like `sudo DOCKER_BUILDKIT=1 docker-compose build` didn't work still producing the same error.

Also now noticed that the buildkit version was added from the 1.25 version. I will try to install the latest docker-compose and let you know the status. 
Thanks! @lmeyerov

**Reply by jwx123** [reply]
you can check the docker status ,maybe docker started but no active

**Reply by kkarthikvk** [reply]
Actually, the issue was with the docker-compose version only. After installing the 1.29.0 version was able to fix that issue.
Now getting the below error :
`=> preparing layers for inline cache                                      1.9s
error: failed to receive status: rpc error: code = Unavailable desc = error reading from server: EOF
ERROR: Service 'streamlit' failed to build : Build failed
`
I was able to run all the commands in docker file. While preparing layers getting the above error.

**Reply by lmeyerov** [reply]
Can you confirm other buildkit-using dockerfiles successfully build? I'm wondering if something is up with the docker install as well..

**Reply by lmeyerov** [reply]
Just checking in -- did that fix?

**Reply by kkarthikvk** [reply]
Hello @lmeyerov ,
Sorry for the delayed reply. I was able to build the image. The issue was with the docker compose. 
But while running `sudo docker-compose up` getting the attached error.
![Screenshot from 2022-10-30 12-41-27](https://user-images.githubusercontent.com/56584548/198866879-b89ff988-5153-433b-89cf-fa7406716c2b.png)
![Screenshot from 2022-10-30 12-41-21](https://user-images.githubusercontent.com/56584548/198866885-7b47f1ef-280e-480e-92ed-0eca89560f63.png)

**Reply by lmeyerov** [reply]
Thanks, will try to reproduce later today! Guessing this is around the optional .toml file support we added recently for easier site-wide theming.

**Reply by lmeyerov** [reply]
@kkarthikvk  Thanks for the heads up on ^^^^ -- just landed a fix to master, you can `git pull` and maybe need a `docker-compose down -v` => `docker-compose build` => `docker-compose up`

**Reply by kkarthikvk** [reply]
Hello @lmeyerov , I'm again facing the docker compose build issue. Looking at it. Will update you shortly. Thanks!

**Comment by lmeyerov** [comment]
hmm not sure

if you are on a cpu box (vs gpu), you may benefit from trying cpu builds (with smaller bases)

```bash
cd docker
./dc.cpu build
./dc.cpu up
```

and what does `docker-compose logs -f -t --tail=100` report?
